pkg-manager/core/test/install/excludeLinksFromLockfile.ts

```diff
@@ -0,0 +1,109 @@
+          'external-1': `link:${path.relative(project1Dir, externalPkg1)}`,
+        },
+      },
+      rootDir: project1Dir,
+    },
+    {
+      buildIndex: 0,
+      manifest: {
+        name: 'project-2',
+        version: '1.0.0',
+
+        dependencies: {
+          'is-negative': '1.0.0',
+          'external-2': `link:${path.relative(project2Dir, externalPkg2)}`,
+        },
+      },
+      rootDir: project2Dir,
+    },
+  ]
+  await mutateModules(importers, await testDefaults({ allProjects, excludeLinksFromLockfile: true }))
+  const lockfile: LockfileV6 = await readYamlFile(WANTED_LOCKFILE)
+  expect(lockfile.importers['project-1'].dependencies?.['external-1']).toBeUndefined()
+  expect(lockfile.importers['project-2'].dependencies?.['external-2']).toBeUndefined()
+
+  expect(fs.existsSync(path.resolve('project-1/node_modules/external-1'))).toBeTruthy()
+  expect(fs.existsSync(path.resolve('project-2/node_modules/external-2'))).toBeTruthy()
+
+  await rimraf('node_modules')
+  await rimraf('project-1/node_modules')
+  await rimraf('project-2/node_modules')
+
+  await mutateModules(importers, await testDefaults({ allProjects, excludeLinksFromLockfile: true, frozenLockfile: true }))
+  expect(lockfile.importers['project-1'].dependencies?.['external-1']).toBeUndefined()
+  expect(lockfile.importers['project-2'].dependencies?.['external-2']).toBeUndefined()
+
+  expect(fs.existsSync(path.resolve('project-1/node_modules/external-1'))).toBeTruthy()
+  expect(fs.existsSync(path.resolve('project-2/node_modules/external-2'))).toBeTruthy()
+
+  await rimraf('node_modules')
+  await rimraf('project-1/node_modules')
+  await rimraf('project-2/node_modules')
+
+  await mutateModules(importers, await testDefaults({ allProjects, excludeLinksFromLockfile: true, frozenLockfile: false, preferFrozenLockfile: false }))
+  expect(lockfile.importers['project-1'].dependencies?.['external-1']).toBeUndefined()
+  expect(lockfile.importers['project-2'].dependencies?.['external-2']).toBeUndefined()
+
+  expect(fs.existsSync(path.resolve('project-1/node_modules/external-1'))).toBeTruthy()
+  expect(fs.existsSync(path.resolve('project-2/node_modules/external-2'))).toBeTruthy()
+
+  delete allProjects[1].manifest.dependencies!['external-2']
+  allProjects[1].manifest.dependencies!['external-3'] = `link:${path.relative(project2Dir, externalPkg3)}`
+  await mutateModules(importers, await testDefaults({ allProjects, excludeLinksFromLockfile: true }))
+  expect(lockfile.importers['project-1'].dependencies?.['external-1']).toBeUndefined()
+  expect(lockfile.importers['project-2'].dependencies?.['external-2']).toBeUndefined()
+  expect(lockfile.importers['project-2'].dependencies?.['external-3']).toBeUndefined()
+
+  expect(fs.existsSync(path.resolve('project-1/node_modules/external-1'))).toBeTruthy()
+  // expect(fs.existsSync(path.resolve('project-2/node_modules/external-2'))).toBeFalsy() // Should we remove external links that are not in deps anymore?
```

Comment:
I think it's necessary to remove outdated links, we need to keep its idempotency